admin/services/pupilService.js

```diff
@@ -0,0 +1,32 @@
+const mongoose = require('mongoose')
+
+const School = require('../models/school')
+const Pupil = require('../models/pupil')
+const Answer = require('../models/answer')
+
+const pupilService = {
+  fetchPupilsData: async (schoolId) => {
+    const [ schoolData, pupils ] = await Promise.all([
+      School.findOne({'_id': schoolId}).lean().exec(),
+      Pupil.getPupils(schoolId).exec()
+    ]).catch((error) => {
+      throw new Error(error)
+    })
+    return {
+      schoolData,
+      pupils
+    }
+  },
+  fetchPupilAnswers: async (id, next) => {
```

Comment:
I don't think a service function should be making decisions like calling `next()` Better just to throw.